src/theory/booleans/circuit_propagator_proofs.h

```diff
@@ -0,0 +1,580 @@
+/*********************                                                        */
+/*! \file circuit_propagator_proofs.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Gereon Kremer
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Proofs for the non-clausal circuit propagator.
+ **
+ ** Proofs for the non-clausal circuit propagator.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__BOOLEANS__CIRCUIT_PROPAGATOR_PROOFS_H
+#define CVC4__THEORY__BOOLEANS__CIRCUIT_PROPAGATOR_PROOFS_H
+
+#include <memory>
+
+#include "expr/node.h"
+#include "expr/proof_node.h"
+#include "theory/booleans/circuit_propagator.h"
+
+namespace CVC4 {
+namespace theory {
+namespace booleans {
+
+namespace {
+
+template <typename T>
+Node mkRat(T val)
+{
+  auto* nm = NodeManager::currentNM();
+  return nm->mkConst<Rational>(val);
+}
+
+inline std::vector<Node> collectButHoldout(TNode parent,
+                                           TNode::iterator holdout,
+                                           bool negate = false)
+{
+  std::vector<Node> lits;
+  for (TNode::iterator i = parent.begin(), i_end = parent.end(); i != i_end;
+       ++i)
+  {
+    if (i != holdout)
+    {
+      lits.emplace_back(negate ? (*i).notNode() : Node(*i));
+    }
+  }
+  return lits;
+}
+
+}  // namespace
+
+struct CircuitPropagatorProver
```

Comment:
ProofCircuitPropagator would follow our naming conventions better.  Also this should be a class (and cleaned up later...).

Reply:
Yes, there will be cleanup. Also adding a cpp, etc. (it is only included by the circuit_propagator.cpp, so it's not so bad having everything in the header for the meantime)